cesmcatalog/gen_CESM_catalog.py

```diff
@@ -0,0 +1,224 @@
+#!/usr/bin/env python
+"""
+  Goal: a script that gets pointed to a CESM case directory and generates
+        an intake-esm catalog of output in the short-term archive directory
+
+  Current state: a script that gets pointed to a CESM case directory and prints
+                 some environment variables that will be necessary for above
+
+  NOTE: this script needs to be run in the CESM postprocessing python environment,
+        which is still based on python 2.7. Follow instructions at
+
+        https://github.com/NCAR/CESM_postprocessing/wiki/cheyenne-and-DAV-quick-start-guide
+"""
+
+import argparse
+import fnmatch
+import logging
+import os
+import subprocess
+import sys
+
+################################################################################
+
+
+def _parse_args():
+    """ Wrapper for argparse, returns dictionary of arguments """
+
+    parser = argparse.ArgumentParser(
+        description='Generate intake-esm catalog for a CESM case',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+
+    parser.add_argument(
+        '-c',
+        '--case-root',
+        action='store',
+        dest='case_root',
+        required=True,
+        help='CESM case root to generate intake-esm catalog for',
+    )
+    parser.add_argument(
+        '-d',
+        '--debug',
+        action='store_true',
+        dest='debug',
+        help='Add additional output for debugging',
+    )
+
+    return parser.parse_args()
+
+
+################################################################################
+
+
+def _find_data(case_root):
+    logger = logging.getLogger(__name__)
+
+    # 1. change directory to case_root (if it exists)
+    try:
+        os.chdir(case_root)
+    except:
+        # TODO: set up logger instead of print statements
+        logger.error('{} does not exist'.format(case_root))
+        sys.exit(1)
```

Comment:
We may want to replace this with a contextmanager approach:

```python
from contextlib import contextmanager
@contextmanager
def chdir(path):
    """
    Change working directory to `path` and restore it again
    This context manager is useful if `path` stops existing during your
    operations.
    """
    old_dir = os.getcwd()
    os.chdir(path)
    try:
        yield
    finally:
        os.chdir(old_dir)
```

You can then use it as follows:

```python
with chdir(case_root):
   ... DO SOME WORK
```

Reply:
Nice, I'll look into this tomorrow (comment is "outdated" but hasn't been resolved yet)